xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/login.vm

```diff
@@ -58,48 +59,46 @@
 <dl>
   <dt>
     <label for="j_username">$services.localization.render('username')</label>
-    #set ($forgotUsernameURL = $services.security.authentication.getAuthenticationURL('retrieveusername', $NULL))
-    #if("$!forgotUsernameURL" != "")
-      <span class="xAdditional"><a href="$forgotUsernameURL" tabindex="500">$services.localization.render('xe.admin.forgotUsername.loginMessage')</a></span>
-    #end
   </dt>
   <dd class="input-group">
     <span class="input-group-addon">$services.icon.renderHTML('user')</span>
-    <input class="form-control" type="text" id="j_username" size="60" name="j_username" value="" tabindex="100"/>
+    ## Ensure that the username field of the login form has the focus to make it easy for users to log in quickly
+    <input class="form-control" type="text" id="j_username" size="60" name="j_username" value="" autofocus />
   </dd>
   <dt>
     <label for="j_password">$services.localization.render('password')</label>
-    #set ($resetPasswordUrl = $services.security.authentication.getAuthenticationURL('resetpassword', $NULL))
-    #if("$!resetPasswordUrl" != '')
-      <span class="xAdditional"><a href="$resetPasswordUrl" tabindex="600">$services.localization.render('xe.admin.passwordReset.loginMessage')</a></span>
-    #end
   </dt>
   <dd class="input-group">
     <span class="input-group-addon">$services.icon.renderHTML('lock')</span>
-    <input type="password" id="j_password" size="60" name="j_password" value="" tabindex="200"/>
+    <input type="password" id="j_password" size="60" name="j_password" value="" />
   </dd>
   <dt></dt>
   <dd>
     <label class="xHint" for="rememberme">
-      <input id="rememberme" type="checkbox" name="j_rememberme" value="true" tabindex="300"/> $services.localization.render('remembermeonthiscomp')
+      <input id="rememberme" type="checkbox" name="j_rememberme" value="true" /> $services.localization.render('remembermeonthiscomp')
     </label>
   </dd>
   #if ("$!supplementaryForm" != "")
     <div id="dedicated-auth-failures-field">
       $supplementaryForm
     </div>
   #end
-  <dt><input type="submit" class="btn btn-primary col-xs-12" value="$services.localization.render('login')" tabindex="400"/></dt>
+  <dt class="submitSection">
+    <input type="submit" class="btn btn-primary col-xs-12" value="$services.localization.render('login')"/>
+
+    #set ($forgotUsernameURL = $services.security.authentication.getAuthenticationURL('retrieveusername', $NULL))
+    #if("$!forgotUsernameURL" != "")
+      <span class="xAdditional"><a href="$forgotUsernameURL">$services.localization.render('xe.admin.forgotUsername.loginMessage')</a></span>
+    #end
+    #set ($resetPasswordUrl = $services.security.authentication.getAuthenticationURL('resetpassword', $NULL))
+    #if("$!resetPasswordUrl" != '')
+      <span class="xAdditional recoverPassword"><a href="$resetPasswordUrl" >$services.localization.render('xe.admin.passwordReset.loginMessage')</a></span>
```

Comment:
The result of this translation should probably be XML escaped.

Reply:
Same for all the above.